esmvaltool/diag_scripts/hydrology/globwat.py

```diff
@@ -0,0 +1,346 @@
+    return cube
+
+
+def langbein_pet(tas):
+    """Calculate potential ET using Langbein method.
+
+    The Langbein curve represents an empirical relationship between temperature
+    and potential ET (pet). Where T is the annual average temperature in degree
+    Celcius, pet is in mm per year, and a, b, c are unitless empirical
+    constants.
+    Reference: https://doi.org/10.3133/cir52 page 8, figure 1.
+    An example of using Langbein method can be found at:
+    https://doi.org/10.1080/02626667.2017.1332416 page 1472, equation 7.
+    """
+    tas.convert_units('degC')
+    constant_a = iris.coords.AuxCoord(np.float32(325),
+                                      long_name='first constant', units=None)
+    constant_b = iris.coords.AuxCoord(np.float32(21),
+                                      long_name='second constant', units=None)
+    constant_c = iris.coords.AuxCoord(np.float32(0.9),
+                                      long_name='third constant', units=None)
+
+    # assumption here: tas is constant over time, then the monthly/daily
+    # average value is equal to the annual average.
+    pet = (tas) * constant_b + (tas ** 2) * constant_c + constant_a
+    pet.units = 'kg m-2 year-1'  # equivalent to mm year-1
+    pet.convert_units('kg m-2 s-1')  # convert to a cmor compatible unit
+    pet.var_name = 'evspsblpot'
+    pet.standard_name = 'water_potential_evaporation_flux'
+    pet.long_name = 'Potential Evapotranspiration'
+    return pet
+
+
+def get_cube_time_info(cube):
+    """Return year, month and day from the cube."""
+    coord_time = cube.coord('time')
+    time = coord_time.cell(0).point
+    time_step = time.strftime("%Y%m%d")
+    return time_step
+
+
+def get_cube_data_info(cube):
+    """Return short_name, and mip from the cube."""
+    short_name = cube.var_name
+    mip = cube.attributes['mip']
+    return short_name, mip
+
+
+def _reindex_data(cube, target):
+    """Reindex data to a global coordinates set.
+
+    Globwat works with global extent.
+    """
+    array = xr.DataArray.from_iris(cube).to_dataset()
+    target_ds = xr.DataArray.from_iris(target).to_dataset()
+    reindex_ds = array.reindex(
+        {"lat": target_ds["lat"], "lon": target_ds["lon"]},
+        method="nearest",
+        tolerance=1e-2,
+    )
+    new_cube = reindex_ds.to_array().to_iris()
+    new_cube.var_name = cube.var_name
+    new_cube.attributes = cube.attributes
+    return new_cube
+
+
+def _swap_western_hemisphere(cube):
+    """Set longitude values in range -180, 180.
+
+    Western hemisphere longitudes should be negative.
+    """
+    array = xr.DataArray.from_iris(cube)
+
+    # Set longitude values in range -180, 180.
+    array['lon'] = (array['lon'] + 180) % 360 - 180
+
+    # Re-index data along longitude values
+    west = array.where(array.lon < 0, drop=True)
+    east = array.where(array.lon >= 0, drop=True)
+    return west.combine_first(east)
+
+
+def _flip_vertically(array):
```

Comment:
maybe just call it `_flip_latitudes`? "Vertically" to me sounds like the height dimension.